core/templates/services/html-length.service.ts

```diff
@@ -126,19 +147,23 @@ export class HtmlLengthService {
       nonTextNode: string, calculationType: CalculationType): number {
     if (nonTextNode.includes('oppia-noninteractive-math')) {
       return 1;
-    }
-    const altTextMatch = nonTextNode
-      .match(/alt-with-value="&amp;quot;([^&]*)&amp;quot;"/);
-
-    let text = [];
-    if (altTextMatch && altTextMatch[1]) {
-      const altText = altTextMatch[1];
-      if (calculationType === CALCULATION_TYPE_WORD) {
-        text = altText.trim().split(' ');
-      } else if (calculationType === CALCULATION_TYPE_CHARACTER) {
-        text = altText.trim().split(' ').join('').split('');
+    } else if (nonTextNode.includes(
+      'oppia-noninteractive-collapsible') || nonTextNode.includes(
+      'oppia-noninteractive-tabs')) {
+      return 1000;
+    } else if (nonTextNode.includes('oppia-noninteractive-video')) {
+      return 0;
+    } else {
```

Comment:
The top-level of classification should always be by the node type, and then you should have specific logic for each node type. This makes the code easier to maintain.

Reply:
The code should be formatted as "if the node is of type X, do this, elseif the node is of type Y, do this, elif the node is of type Z, do this". The code in this else block does not follow that logic because it is not conditioning on a particular RTE component type at the outset.

Reply:
Done! Thanks!